src/core/libmaven/mzUtils.cpp

```diff
@@ -593,93 +426,90 @@ Series:  Prentice-Hall Series in Automatic Computation
         for (int i = 0; i < n; i++) {
             sumx += x[i];
             sumy += y[i];
-            sumxy += x[i]*y[i];
-            x2 += x[i]*x[i];
-            y2 += y[i]*y[i];
+            sumxy += x[i] * y[i];
+            x2 += x[i] * x[i];
+            y2 += y[i] * y[i];
         }
         if (n == 0) return 0;
-        double var1 = x2-(sumx*sumx)/n;
-        double var2 = y2-(sumy*sumy)/n;
+        double var1 = x2 - (sumx * sumx) / n;
+        double var2 = y2 - (sumy * sumy) / n;
         if ( var1 == 0 || var2 == 0 ) return 0;
-        return (sumxy -( sumx*sumy)/n) / sqrt((x2-(sumx*sumx)/n)*(y2-(sumy*sumy)/n));
+        float corr = (sumxy -( sumx * sumy) / n) /
+                       sqrt((x2 - (sumx * sumx) / n) *
+                          (y2 - (sumy * sumy) / n));
+        return corr;
     }
 
-
-    /*peak fitting function*/
-    void gaussFit(const vector<float>&ycoord, float* sigma, float* R2) {
-
+    pair<float, float> gaussFit(const vector<float>& ycoord)
+    {
         float s = 20;
         float min_s = 0;
         float minR = 1e99;
 
         //find best fit
-        if (ycoord.size()<3) return;
-        vector<float>yobs=ycoord;
-        int ysize=yobs.size();
-        int midpoint  = int(ysize/2);
+        if (ycoord.size()<3){
+            pair<float, float> res = make_pair(FLT_MAX, FLT_MAX);
```

Comment:
Can we use `numeric_limits<float>::max()` for the maximum allowed value for floating point values. `FLT_MAX` is a C thing. Also use `numeric_limits` whenever you need to obtain the min/max/lowest representable values for any numeric type.